cdc/scheduler/info_provider.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
```

Comment:
why not add the file to the schedule_dispatcher.go file

Reply:
I was thinking of improving readability by separating non-core logic into a separate file. Moreover, we might want to refactor the relevant logic into a separate struct implementing `InfoProvider`.